pg_hint_plan.c

```diff
@@ -1014,7 +1016,7 @@ HintStateDelete(HintState *hstate)
 	if (hstate->hint_str)
 		pfree(hstate->hint_str);
 
-	for (i = 0; i < hstate->num_hints[HINT_TYPE_SCAN_METHOD]; i++)
+	for (i = 0; i < hstate->nall_hints; i++)
```

Comment:
This one has been already fixed in PG14~, but it also looks that this could lead to incorrect memory lookups depending on the state of the stack, so I have fixed that separately down to 10.